Program.cs

```diff
@@ -1,4 +1,38 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// REVERSE NUMBER GUESSING GAME (LSD TO MSD)
+// Write the code for displaying the number you guessed.
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Linq.Expressions;
 
-   
\ No newline at end of file
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>Reverse Guessing Game</summary>
+   internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>insert the user to think a number between 1 and 100</summary>
+      /// <param name="args">arguments</param>
+      static void Main (string[] args) {
+         Console.WriteLine ("Welcome to guessing game.");
+         Console.WriteLine ("Think a number between 1 and 100.");
+         int rem = 0; string binary = "";
+         for (int i = 1; i <= 7; i++) {
+            Console.Write ($"Is the remainder {rem} when divided by {Math.Pow (2, i)}: (y)es or (n)o: ");//initially reminder will be 0.
+            string userinput = Console.ReadLine ();
+            if (userinput == "y")
+               binary = "0" + binary;//add binary number in the binary string.
+            else if (userinput == "n") {

```

Comment:
Use Console.ReadKey(); to read the key input from the user.
Your intention here is to get the input, which is very clear either 'Y' or 'N', and anything other than this key is invalid input.
So, use [ConsoleKey](https://learn.microsoft.com/en-us/dotnet/api/system.consolekey?view=net-7.0) to validate the input.